src/NUnitFramework/framework/Internal/ParamAttributeTypeConversions.cs

```diff
@@ -104,22 +115,33 @@ public static bool TryConvert(object value, Type targetType, out object converte
 
             bool convert = false;
 
-            if (targetType == typeof(short) || targetType == typeof(byte) || targetType == typeof(sbyte))
+            if (targetType == typeof(short) || targetType == typeof(byte) || targetType == typeof(sbyte) || targetType == typeof(long?) ||
+                targetType == typeof(short?) || targetType == typeof(byte?) || targetType == typeof(sbyte?) || targetType == typeof(double?))
```

Comment:
I'd prefer to use `Nullable.GetUnderlyingType(targetType)` to remove redundancy.